src/document/doctree.ml

```diff
@@ -283,6 +283,35 @@ end = struct
     |> snd
 end
 
+module PageTitle : sig 
+  val render_title : Page.t -> Item.t list
+end = struct
+  let format_title kind name=
+    let mk title =
+      let level = 0 and label = None in
+      [ Types.Item.Heading { level; label; title } ]
+    in
+    let prefix s =
+      mk (Types.inline (Text (s ^ " ")) :: Codefmt.code (Codefmt.txt name)) in
+    match kind with
+    | `BaseModule -> prefix "Module"
+    | `Module -> prefix "Module"
+    | `Argument -> prefix "Parameter"
+    | `ModuleType -> prefix "Module type"
+    | `ClassType -> prefix "Class type"
+    | `Class -> prefix "Class"
+    | `Page -> []
```

Comment:
Aren't you dropping the page name here ?

Reply:
The page title is embedded in the page it seems.

Reply:
I had the tests fail with a duplicate title when I kept the code as it was. The reason is that it was dead code as the title was never rendered into the header anyways, in the case of `Page` (see line 1697 here): 
https://github.com/ocaml/odoc/blob/a52dd8e960f99b38162d40311a9c4512d5741d43/src/document/generator.ml#L1691-L1698